static/js/stream_create.js

```diff
@@ -278,27 +279,49 @@ export function show_new_stream_modal() {
     $("#stream-creation").removeClass("hide");
     $(".right .settings").hide();
 
-    const html = blueslip.measure_time("render new stream users", () => {
-        const all_users = people.get_people_for_stream_create();
-        // Add current user on top of list
-        all_users.unshift(people.get_by_user_id(page_params.user_id));
-        return render_new_stream_users({
-            users: all_users,
+    const add_people_container = $("#people_to_add");
+    add_people_container.html(
+        render_new_stream_users({
             streams: stream_settings_data.get_streams_for_settings_page(),
-            is_admin: page_params.is_admin,
-        });
+        }),
+    );
+
+    all_users = people.get_people_for_stream_create();
+    // Add current user on top of list
+    const current_user = people.get_by_user_id(page_params.user_id);
+    all_users.unshift({
+        email: current_user.email,
+        user_id: current_user.user_id,
+        full_name: current_user.full_name,
+        checked: true,
+        disabled: !page_params.is_admin,
+    });
+
+    all_users_list_widget = ListWidget.create($("#user-checkboxes"), all_users, {
+        name: "new_stream_add_users",
+        parent_container: add_people_container,
+        modifier(item) {
+            return render_new_stream_user(item);
+        },
+        filter: {
+            element: $("#people_to_add .add-user-list-filter"),
+            predicate(item, value) {
+                return item.full_name.toLowerCase().includes(value);
+            },
+        },
+        simplebar_container: $("#user-checkboxes-simplebar-wrapper"),
+        html_selector: (user) => $(`#${CSS.escape("user_checkbox_" + user.user_id)}`),
```

Comment:
Do we need this, or could we just use `data-user-id` for the selector?  We're generally trying to move away from constructing HTML IDs as a string this way.

Reply:
Maybe worth testing with a profiler with this specific case?  I'd expect it to not make much difference with a properly scoped selector.